analytics/src/analytics/cli.py

```diff
@@ -106,23 +98,39 @@ def calculate_deliverable_percent_complete(
     post_results: Annotated[bool, POST_RESULTS_ARG] = False,
 ) -> None:
     """Calculate percentage completion by deliverable."""
-    # defer load of settings until this command is called
-    # this prevents an error if ANALYTICS_SLACK_BOT_TOKEN env var is unset
-    from config import settings
-
     # load the input data
     task_data = DeliverableTasks.load_from_json_files(
         sprint_file=sprint_file,
         issue_file=issue_file,
     )
-    # calculate burndown
-    pct_complete = DeliverablePercentComplete(task_data, unit=unit)
+    # calculate percent complete
+    metric = DeliverablePercentComplete(task_data, unit=unit)
+    show_and_or_post_results(
+        metric=metric,
+        show_results=show_results,
+        post_results=post_results,
+    )
+
+
+def show_and_or_post_results(
+    metric: BaseMetric,
+    *,  # makes the following args keyword only
+    show_results: bool,
+    post_results: bool,
+) -> None:
+    """Optionally show the results of a metric and/or post them to slack."""
+    # defer load of settings until this command is called
+    # this prevents an error if ANALYTICS_SLACK_BOT_TOKEN env var is unset
+    from config import settings
+
     # optionally display the burndown chart in the browser
     if show_results:
-        pct_complete.show_chart()
+        metric.show_chart()
+        print("Slack message:\n")
```

Comment:
should these `print`s be logging statements? I don't have a strong opinion on this, just asking.

Reply:
Good suggestion! I'll work on this in this ticket: #742